### PR TITLE
Previous AK fixes applied in xdmod7.0 branch

### PR DESCRIPTION
**Pull request by nsimakov**
copying previously applied fixes in xdmod7.0 branch

### CODE REVIEW

File: build.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "xdmod-appkernels",
-    "version": "7.0.0",
```
Comment (plessbd) [inline build.json]: this is probably going to need to be changed to be 7.5

File: html/controllers/data_explorer/common.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}
+    
+function getExpandedAppKernels()
+{
+    return isset($_REQUEST['expandedAppKernels'])  && $_REQUEST['expandedAppKernels'] != '' ? explode(',', $_REQUEST['expandedAppKernels']):array();
+}
+    
+function getShowChangeIndicator()
+{
+    return  ( isset($_REQUEST['show_change_indicator'])
```
Comment (plessbd) [inline html/controllers/data_explorer/common.php]: extra space between return and (

File: html/controllers/data_explorer/get_tree.php [hunk 3]
```diff
@@ -0,0 +1,191 @@
+<?php
+@require_once 'common.php';
```
Comment (plessbd) [inline html/controllers/data_explorer/get_tree.php]: why is this preventing errors from being thrown?

File: html/controllers/data_explorer/get_tree.php [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+try
+{
+    
+    $user = \xd_security\getLoggedInUser();
+   
+    $ak_db = new \AppKernel\AppKernelDb();
+     
+    if(isset($_REQUEST['node']) && $_REQUEST['node'] === 'resources') {
+        $selectedResourceIds = getSelectedResourceIds();
+        $selectedProcessingUnits = array();//getSelectedPUCounts();
```
Comment (plessbd) [inline html/controllers/data_explorer/get_tree.php]: what is the point of this comment?

File: html/controllers/data_explorer/get_tree.php [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        $returnData = array('totalCount'=> 1, 'data' => array(array('nodes' => json_encode(array_values($pus)))));
+
+    }
+    elseif(isset($_REQUEST['node']) && $_REQUEST['node'] === 'app_kernels') {
+        $selectedResourceIds = getSelectedResourceIds();
+        $selectedMetrics = getSelectedMetrics();
+        $expandedAppKernels = getExpandedAppKernels();
+        checkDateParameters();
+                
+        //$app_kernels = $ak_db->getUniqueAppKernels('2010-01-01',date("Y-m-d")/*,
```
Comment (plessbd) [inline html/controllers/data_explorer/get_tree.php]: remove commented out code.

File: html/controllers/data_explorer/get_tree.php [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+        //                                            $selectedResourceIds, $selectedProcessingUnits*/);
+            
+        $all_app_kernels = $ak_db->getUniqueAppKernels();
+        foreach($all_app_kernels as $app_kernel)
+        {
+            $metrics = $ak_db->getMetrics(
+                $app_kernel->id,
+                '2010-01-01',
+                date("Y-m-d")/*,
+                $selectedResourceIds, $selectedProcessingUnits*/
```
Comment (plessbd) [inline html/controllers/data_explorer/get_tree.php]: remove commented out code

File: html/controllers/data_explorer/get_tree.php [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+                    'start_ts' => $app_kernel->start_ts,
+                    'end_ts' => $app_kernel->end_ts,
+                    'ak_def_id' => $app_kernel->id,
+                    'type' => 'pu',
+                    'iconCls' => 'node',
+                    'leaf' => true,
+                    //'disabled' => $metric_disabled,
+                    'uiProvider' => 'Ext.tree.TriStateNodeUI',
+                    'checked' =>  in_array($c_id.'_'.$pu->count, $selectedMetrics) || in_array($c_id, $selectedMetrics)
+                    //'children' => $pu_children
```
Comment (plessbd) [inline html/controllers/data_explorer/get_tree.php]: remove commented out code

File: html/controllers/data_explorer/get_tree.php [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+                $children[] = array('text' => $metric->name,
+                    'id' => $c_id,
+                    'qtip' => $metric->name,
+                    'start_ts' => $app_kernel->start_ts,
+                    'end_ts' => $app_kernel->end_ts,
+                    'ak_def_id' => $app_kernel->id,
+                    'type' => 'metric',
+                    'iconCls' => 'metric',
+                    'leaf' => false,
+                    //'singleClickExpand' => true,
```
Comment (plessbd) [inline html/controllers/data_explorer/get_tree.php]: remove commented out code

File: html/controllers/data_explorer/get_tree.php [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+                    'expanded' => in_array($c_id, $expandedAppKernels),
+                    'uiProvider' => 'Ext.tree.TriStateNodeUI',
+                    //'disabled' => $metric_disabled,
```
Comment (plessbd) [inline html/controllers/data_explorer/get_tree.php]: remove commented out code

File: html/controllers/data_explorer/get_tree.php [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+                    'checked' => in_array($c_id, $selectedMetrics),
+                    'children' => $pu_children
+                    );
+            }
+            $returnData[] = array('text' => $app_kernel->name,/*.' '.date('Y-m-d',$app_kernel->start_ts).' '.date('Y-m-d', $app_kernel->end_ts)*/
```
Comment (plessbd) [inline html/controllers/data_explorer/get_tree.php]: remove commented out code

File: html/gui/js/modules/app_kernels/AppKernelNotificationPanel.js [hunk 11]
```diff
@@ -8,10 +8,11 @@ XDMoD.Module.AppKernels.AppKernelNotificationPanel = Ext.extend(Ext.Panel, {
             var form = this.notificationSettingsForm.getForm();
             var formData = Ext.encode(form.getValues());
 
-            if (formData == '{}')
+            if (formData === '{}') {
                 formData = '{"send_report_weekly_on_day":"Monday","send_report_monthly_on_day":"1","resourcesList_all":"on","appkernelsList_all":"on","controlThreshold":"-0.5"}';
+            }
 
-            /*form.load({
+            /* form.load({
```
Comment (plessbd) [inline html/gui/js/modules/app_kernels/AppKernelNotificationPanel.js]: remove commented out code

File: html/gui/js/modules/app_kernels/AppKernelNotificationPanel.js [hunk 12]
```diff
@@ -148,39 +141,39 @@ XDMoD.Module.AppKernels.AppKernelNotificationPanel = Ext.extend(Ext.Panel, {
             restful: true,
             listeners: {
                 scope: this,
-                load: function (t, records, options) {
+                load: function (t, records) {
                     var i = 0;
                     for (i = 0; i < records.length; i++) {
-                        //console.log(records[i].data.name);
-                        /*var h = new Ext.form.Hidden({
+                        // console.log(records[i].data.name);
```
Comment (plessbd) [inline html/gui/js/modules/app_kernels/AppKernelNotificationPanel.js]: remove commented out code

File: html/gui/js/modules/app_kernels/AppKernelNotificationPanel.js [hunk 13]
```diff
@@ -148,39 +141,39 @@ XDMoD.Module.AppKernels.AppKernelNotificationPanel = Ext.extend(Ext.Panel, {
...
+                    // fireEvent(
+                    // this.loadSettings();
                 }
             }
         });
-        //this.resourcesListStore.load();
-        //this.resourcesListStore.save( );
-        //appkernelsList
+        // this.resourcesListStore.load();
+        // this.resourcesListStore.save( );
```
Comment (plessbd) [inline html/gui/js/modules/app_kernels/AppKernelNotificationPanel.js]: remove commented out code

File: html/gui/js/modules/app_kernels/AppKernelNotificationPanel.js [hunk 14]
```diff
@@ -225,21 +218,21 @@ XDMoD.Module.AppKernels.AppKernelNotificationPanel = Ext.extend(Ext.Panel, {
             ['sendOnPatternRecFailedRuns', 'Send on Major Pattern Errors']
         ];
         this.notificationSettingsForm = new Ext.form.FormPanel({
-            //xtype:'fieldset',
-            //title:'Report Periodicity Settings',
+            // xtype:'fieldset',
```
Comment (plessbd) [inline html/gui/js/modules/app_kernels/AppKernelNotificationPanel.js]: remove commented out code

File: html/gui/js/modules/app_kernels/AppKernelNotificationPanel.js [hunk 15]
```diff
@@ -564,14 +556,15 @@ XDMoD.Module.AppKernels.AppKernelNotificationPanel = Ext.extend(Ext.Panel, {
             ]
         });
         XDMoD.Module.AppKernels.AppKernelNotificationPanel.superclass.initComponent.apply(this, arguments);
-        //XDMoD.Arr.NotificationPanel.superclass.constructor.call(this, config);
+        // XDMoD.Arr.NotificationPanel.superclass.constructor.call(this, config);
 
-        //this.appkernelsListStore.load();
-        //this.resourcesListStore.load();
+        // this.appkernelsListStore.load();
```
Comment (plessbd) [inline html/gui/js/modules/app_kernels/AppKernelNotificationPanel.js]: remove commented out code

File: html/controllers/data_explorer/get_tree.php (hunk 4 above)
Comment (nsimakov) [reply html/controllers/data_explorer/get_tree.php]: done

**Pull request merged**